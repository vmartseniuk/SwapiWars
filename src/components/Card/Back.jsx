import React, { memo } from "react";
/* eslint-disable react/prop-types */
import { BackSide } from "react-flippy";
import makeStyles from "@material-ui/core/styles/makeStyles";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
/* @core */
import colors from "config/assets";
import convertToRoman from "utils/toRoman";

const useStyles = makeStyles(() => ({
  front: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#202020",
    borderRadius: 20,
    boxShadow: `0 3px 20px -8px rgba(${colors.whiteRGB}, 0.42), 0 4px 5px 0px rgba(${colors.whiteRGB}, 0.12), 0 3px 1px -5px rgba(${colors.whiteRGB}, 0.2)`,
  },
  title: {
    fontSize: "16px",
    textAlign: "center",
    letterSpacing: "5px",
    textTransform: "uppercase",
  },
  opening: {
    padding: "15px 0",
    fontSize: "14px",
    textTransform: "uppercase",
    textAlign: "center",
  },
  button: {
    fontWeight: 300,
    fontSize: "12px",
    lineHeight: "14px",
    letterSpacing: "2px",
    textTransform: "none",
    background: "#272727",
  },
}));

const Back = (props) => {
  const { opening_crawl: opening, episode_id: episode, onClick } = props;
  const classes = useStyles();

  return (
    <BackSide className={classes.front}>
      <Typography className={classes.title}>
        {convertToRoman(episode)}
        &nbsp; episode
      </Typography>

      <Typography className={classes.opening}>{opening}</Typography>
      <Button className={classes.button} variant="outlined" size="large" onClick={onClick}>
        show more
      </Button>
    </BackSide>
  );
};
export default memo(Back);
