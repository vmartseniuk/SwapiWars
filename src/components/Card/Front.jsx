import React, { memo } from "react";
/* eslint-disable react/prop-types */
import { FrontSide } from "react-flippy";
import makeStyles from "@material-ui/core/styles/makeStyles";
import Typography from "@material-ui/core/Typography";
import convertToRoman from "utils/toRoman";
import Trooper from "media/image/trooper.png";
import DarthVader from "media/image/darth-vader.png";
import Yoda from "media/image/yoda.png";
import BobaFett from "media/image/boba-fett.png";
import Amidala from "media/image/amidala.png";

const randomBg = () => {
  const random = Math.floor(Math.random() * 5) + 0;
  const bigSize = [Trooper, DarthVader, Yoda, BobaFett, Amidala];
  return bigSize[random];
};

const useStyles = makeStyles((theme) => ({
  front: {
    backgroundColor: "transparent",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    cursor: "pointer",
  },
  container: {
    height: "inherit",
  },
  background: ({ img }) => ({
    position: "absolute",
    top: "40%",
    left: "50%",
    transform: "translate(-50%,-50%)",
    // background: colors.cardLabelBg,
    backgroundSize: "cover",
    backgroundPosition: "center center",
    backgroundImage: `linear-gradient(rgba(0,0,0,.6), rgba(0,0,0,.6)), url(${img})`,

    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    border: "1px solid white",
    borderRadius: "20px",
    width: "110px",
    height: "110px",
    [theme.breakpoints.up("md")]: {
      width: "130px",
      height: "130px",
    },
  }),
  background_episode: {
    fontSize: 25,
  },
  background_text: {
    fontSize: 15,
    textTransform: "uppercase",
    letterSpacing: "5px",
  },
  title: {
    textAlign: "center",
    display: "inline",
    fontSize: "25px",
    zIndex: 3,
    marginTop: "40%",
  },
}));

const Front = (props) => {
  const { episode_id: episode, title } = props;
  const img = randomBg();
  const classes = useStyles({ img });
  return (
    <FrontSide className={classes.front}>
      <div className={classes.background}>
        <Typography variant="h5" className={classes.background_episode}>
          {convertToRoman(episode)}
        </Typography>
        <Typography variant="h5" className={classes.background_text}>
          episode
        </Typography>
      </div>
      <Typography className={classes.title}>{title}</Typography>
    </FrontSide>
  );
};

export default memo(Front);
