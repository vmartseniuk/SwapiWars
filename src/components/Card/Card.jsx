import React, { useState } from "react";
/* eslint-disable react/prop-types */
import Flippy from "react-flippy";
import FilmModal from "modals/FilmModal/FilmModal";

import Front from "./Front";
import Back from "./Back";

const style = { width: "320px", height: "400px", margin: "0 auto" };

const Card = (props) => {
  const { info } = props;
  const [open, setOpen] = useState(false);

  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  const matches = window.innerWidth < 600;
  return (
    <>
      <Flippy flipOnHover={!matches} flipOnClick={matches} flipDirection="horizontal" style={style}>
        <Front {...info} />
        <Back {...info} onClick={handleOpen} />
      </Flippy>
      {open && <FilmModal {...info} open={open} onClose={handleClose} />}
    </>
  );
};
export default Card;
