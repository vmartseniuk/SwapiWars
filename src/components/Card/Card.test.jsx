import { create } from "react-test-renderer";
import Card from "./Card";

describe("Card", () => {
  it("renders correctly", () => {
    const props = {}; //
    const rendered = create(<Card {...props}>test name</Card>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
