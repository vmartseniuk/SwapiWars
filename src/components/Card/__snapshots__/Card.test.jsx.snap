// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="flippy-container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "height": "400px",
      "margin": "0 auto",
      "width": "320px",
    }
  }
>
  <div
    className="flippy-cardContainer-wrapper horizontal"
  >
    <div
      className="flippy-cardContainer  "
    >
      <div
        className="flippy-card flippy-front makeStyles-front-1"
        style={
          Object {
            "transitionDuration": "0.6s",
          }
        }
      >
        <div
          className="makeStyles-background-3 makeStyles-background-7"
        >
          <h5
            className="MuiTypography-root makeStyles-background_episode-4 MuiTypography-h5"
          >
            
          </h5>
          <h5
            className="MuiTypography-root makeStyles-background_text-5 MuiTypography-h5"
          >
            episode
          </h5>
        </div>
        <p
          className="MuiTypography-root makeStyles-title-6 MuiTypography-body1"
        />
      </div>
      <div
        className="flippy-card flippy-back makeStyles-front-8"
        style={
          Object {
            "transitionDuration": "0.6s",
          }
        }
      >
        <p
          className="MuiTypography-root makeStyles-title-9 MuiTypography-body1"
        >
          
          Â  episode
        </p>
        <p
          className="MuiTypography-root makeStyles-opening-10 MuiTypography-body1"
        />
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-11 MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            show more
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
